### PR TITLE
data: add export to ICS support

### PR DESCRIPTION
**Pull request by skyzh**
fix https://github.com/SJTU-Plus/course-plus/issues/38

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -96,6 +96,30 @@
...
+                            v-model="firstDayDate"
+                          />
+                        </div>
+                        <div class="col-12">
+                          <button
+                            type="button"
+                            class="btn btn-outline-primary"
+                            @click="downloadAsICS"
+                          >
+                            导出为 ics 文件
```
Comment (LuminousXLB) [inline src/App.vue:117]: 建议修改为导出到iCalendar

**Review by danyang685 — CHANGES_REQUESTED** [review]
I doubt the timetable is outdated. Could you confirm this?

File: src/models/index.ts [hunk 2]
```diff
@@ -93,3 +96,100 @@ export interface LessonDetail {
...
+          time.getDate(),
+          Math.floor(startAt / 60),
+          startAt % 60
+        ],
+        duration: { hours: Math.floor(duration / 60), minutes: duration % 60 },
+        title: `${lesson.kch} ${lesson.kcmc}`,
+        description: `${lesson.jszc}, ${lesson.xf} 学分, 第 ${week} 周`,
+        location: parseTimeLocationDay(lesson.sksj, lesson.jxdd, lesson.xqj)
+      };
+      events.push(event);
```
Comment (LuminousXLB) [inline src/models/index.ts:186]: Would you like to use `recurrenceRule`

File: src/models/index.ts [hunk 3]
```diff
@@ -93,3 +96,100 @@ export interface LessonDetail {
...
+      const duration = endAt - startAt;
+      const event: EventAttributes = {
+        start: [
+          time.getFullYear(),
+          time.getMonth() + 1,
+          time.getDate(),
+          Math.floor(startAt / 60),
+          startAt % 60
+        ],
+        duration: { hours: Math.floor(duration / 60), minutes: duration % 60 },
```
Comment (LuminousXLB) [inline src/models/index.ts:181]: According to the document of ics, 

> `end`: Time at which event ends. Either end or duration is required, but not both.
> https://www.npmjs.com/package/ics

`duration` and `end` should not be set simultaneously

File: src/models/index.ts (hunk 2 above)
Comment (skyzh) [reply src/models/index.ts:186]: recurrenceRule is not easy to use for some courses that takes place on 3, 7, 9, 15 week. I prefer generating one event for one time block, as an event may carry some week-related data (e.g. first week, second week).

File: src/models/index.ts (hunk 3 above)
Comment (skyzh) [reply src/models/index.ts:181]: In fact, I didn't specify end. I just used end time to calculate duration.

**Comment by skyzh** [comment]
> I doubt the timetable is outdated. Could you confirm this?

Except for the 14th course, others should be fine.

File: src/models/index.ts (hunk 2 above)
Comment (LuminousXLB) [reply src/models/index.ts:186]: But it will be crazy if I want to remove them from my calendar one day

File: src/models/index.ts (hunk 3 above)
Comment (LuminousXLB) [reply src/models/index.ts:181]: Oh yes

File: src/models/index.ts (hunk 2 above)
Comment (skyzh) [reply src/models/index.ts:186]: Add a new calendar, import these events. If you want to start again, remove the whole calendar :(

Or we can add an option to use recurring rule in next update

**Review by danyang685 — APPROVED** [review]

**Pull request merged**